syntax = "proto2";

message ImageResizer {
  oneof image_resizer_oneof {
    DefaultResizer default_resizer = 1;
    FixedShapeResizer fixed_shape_resizer = 2;
    KeepAspectRatioResizer keep_aspect_ratio_resizer = 3;
  }
}

message DefaultResizer {
}

message FixedShapeResizer {
  // Desired height of image in pixels.
  optional int32 height = 1 [default = 300];

  // Desired width of image in pixels.
  optional int32 width = 2 [default = 300];
}

message KeepAspectRatioResizer {
  // Desired size of the larger image dimension in pixels.
  optional int32 max_dimension = 1 [default = 600];

  // Whether to pad the image with zeros so the output spatial size is
  // [max_dimension, max_simension].
  optional bool pad_to_max_dimension = 2 [default = false];
}
